dap-mode.el

```diff
@@ -264,6 +264,7 @@ The hook will be called with the session file and the new set of breakpoint loca
   (workspace nil)
   (threads nil)
   (thread-states (make-hash-table :test 'eql) :read-only t)
+  (ui-display-frame nil)
```

Comment:
Obviously the first question is whether this is an appropriate place to store this info.  Second issue is naming?

Reply:
Do we want this so each session can has different display frame? Or if there's only one session, just a global `defvar` is enough?

Reply:
If `dap-mode` can't accommodate multiple simultaneous sessions(?), and isn't ever expected to, then a `defvar` definitely seems more appropriate.  Otherwise I would expect you'd give each its own frame.

I'd be happy to make that change.  I'm just getting started with `dap-mode` as of yesterday and hit this issue.  Let me know, thanks!

Reply:
@WJCFerguson I'm assuming the windows causing issues are the ones created by `dap-ui-many-windows-mode` (which is enabled by default). In that case, your approach may be inappropriate: the `dap-ui-many-windows-mode` windows would be created in the same as the debug session, but others spawned manually would be too. So if someone starts a debug session, switches to a different frame and then does `M-x dap-ui-locals`, the locals window would be displayed in the old frame, not the new one, as one would probably expect.

As to your questions: `dap-mode` supports multiple parallel (running) debug sessions, which can be inspected using `dap-ui-sessions`.

Thanks for contributing to `dap-mode`!